I18n of lttoolbox

ahmedsiam0 opened the pull request.

ahmedsiam0 committed:
Include the i18n library and internationalize lt-proc

mr-martian commented:
`.github/workflows/build.yaml` needs to be updated to install the `i18n` package from `apertium.projectjj.com` - you can probably copy the relevant code from the workflow in one of the other repos that has to install `lttoolbox` to build.

ahmedsiam0 committed:
i18n of lttoolbox patch 1

ahmedsiam0 committed:
i18n of lttoolbox patch 2

ahmedsiam0 committed:
i18n of lttoolbox patch 3

ahmedsiam0 committed:
i18n of lttoolbox patch 4

ahmedsiam0 committed:
i18n of lttoolbox patch 5

ahmedsiam0 committed:
correct error

ahmedsiam0 committed:
correct LTTB1070

ahmedsiam0 committed:
Add i18n library, fix locales issue, make spanish sample

ahmedsiam0 committed:
Add i18n data macro to cxxflags in setup.py

ahmedsiam0 committed:
Change icuformat to formatmsg & update build.yml

ahmedsiam0 committed:
Update error code format

ahmedsiam0 committed:
Add a new line at the end of i18n.h

ahmedsiam0 committed:
simplify i18n.h and add comments to it + some edits to i18n code

ahmedsiam0 committed:
Make format and error member functions in I18n class constant

ahmedsiam0 committed:
Fix a bug related to prompting ALT80640 error in an invalid case.

This bug happened when I was trying to translate a message using apertium and en-eo package
Instead of getting the translated message like in original code, It was prompting ALT80640 error.
